site/app/controllers/AutogradingStatusController.php

```diff
@@ -0,0 +1,72 @@
+<?php
+
+namespace app\controllers;
+
+use app\libraries\Core;
+use app\libraries\response\MultiResponse;
+use app\libraries\response\WebResponse;
+use app\libraries\response\JsonResponse;
+use app\libraries\routers\AccessControl;
+use app\libraries\GradingQueue;
+use app\views\AutogradingStatusView;
+use app\views\ErrorView;
+use Symfony\Component\Routing\Annotation\Route;
+
+class AutogradingStatusController extends AbstractController {
+    /**
+     * @Route("/autograding_status", methods={"GET"})
+     * @return WebResponse | MultiResponse
+     */
+    public function getGradingDonePage() {
+        $instructors = $this->core->getQueries()->getActiveUserIds(true, false, false, false, true);
+        $user_id = $this->core->getUser()->getId();
+        if (!in_array($user_id, $instructors)) {
```

Comment:
Potentially a quicker check could be to call getInstructorLevelAccessCourse and see if there are any results rather than collecting all instructors and faculty.